modules/apps/batch-engine/batch-engine-service/src/main/java/com/liferay/batch/engine/internal/reader/BatchEngineImportTaskItemReaderUtil.java

```diff
@@ -60,13 +62,29 @@ public static <T> T convertValue(
 				}
 			}
 
+			JsonDeserialize[] jsonDeserialize = field.getAnnotationsByType(
+				JsonDeserialize.class);
+
+			Object value = entry.getValue();
+
+			if ((jsonDeserialize.length > 0) &&
+				!Objects.equals(value.getClass(), String.class)) {
+
+				try {
+					value = _objectMapper.writeValueAsString(value);
+				}
+				catch (JsonProcessingException jsonProcessingException) {
+					if (_log.isDebugEnabled()) {
+						_log.debug(jsonProcessingException);
+					}
+				}
+			}
+
 			if (field != null) {
 				field.setAccessible(true);
 
 				field.set(
-					item,
-					_objectMapper.convertValue(
-						entry.getValue(), field.getType()));
+					item, _objectMapper.convertValue(value, field.getType()));
```

Comment:
Do you need to call `convertValue` again once the `JsonDeserialize` element has been used?

Reply:
no longer applies.